tests/slurm_command_gen_strategy/test_common_slurm_command_gen_strategy.py

```diff
@@ -120,3 +120,138 @@ def test_raises_if_no_default_partition(slurm_system: SlurmSystem):
+            [Mock(test=Mock(name="test2", test_template=Mock()))],
+            [
+                "SUCCESS_0=$(grep -q 'Avg bus bandwidth' stdout.txt && echo 1 || echo 0)",
+                "PROLOGUE_SUCCESS=$( [ $SUCCESS_0 -eq 1 ] && echo 1 || echo 0 )",
+                "if [ $PROLOGUE_SUCCESS -eq 1 ]; then",
+                "    srun",
+                "    epilogue",
+                "fi",
+            ],
+        ),
+        # Multiple prologues, multiple epilogues
+        (
+            [Mock(test=Mock(name="test1", test_template=Mock())), Mock(test=Mock(name="test2", test_template=Mock()))],
+            [Mock(test=Mock(name="test3", test_template=Mock())), Mock(test=Mock(name="test4", test_template=Mock()))],
+            [
+                "SUCCESS_0=$(grep -q 'Avg bus bandwidth' stdout.txt && echo 1 || echo 0)",
+                "SUCCESS_1=$(grep -q 'Avg bus bandwidth' stdout.txt && echo 1 || echo 0)",
+                "PROLOGUE_SUCCESS=$( [ $SUCCESS_0 -eq 1 ] && [ $SUCCESS_1 -eq 1 ] && echo 1 || echo 0 )",
+                "if [ $PROLOGUE_SUCCESS -eq 1 ]; then",
+                "    srun",
+                "    epilogue",
+                "    epilogue",
+                "fi",
+            ],
+        ),
+        # Multiple prologues, no epilogue
+        (
+            [Mock(test=Mock(name="test1", test_template=Mock())), Mock(test=Mock(name="test2", test_template=Mock()))],
+            None,
+            [
+                "SUCCESS_0=$(grep -q 'Avg bus bandwidth' stdout.txt && echo 1 || echo 0)",
+                "SUCCESS_1=$(grep -q 'Avg bus bandwidth' stdout.txt && echo 1 || echo 0)",
+                "PROLOGUE_SUCCESS=$( [ $SUCCESS_0 -eq 1 ] && [ $SUCCESS_1 -eq 1 ] && echo 1 || echo 0 )",
+                "if [ $PROLOGUE_SUCCESS -eq 1 ]; then",
+                "    srun",
+                "fi",
+            ],
+        ),
+        # No prologue, multiple epilogues
+        (
+            None,
+            [Mock(test=Mock(name="test3", test_template=Mock())), Mock(test=Mock(name="test4", test_template=Mock()))],
+            [
+                "srun",
+                "epilogue",
+                "epilogue",
+            ],
+        ),
+        # Multiple prologues, single epilogue
+        (
+            [Mock(test=Mock(name="test1", test_template=Mock())), Mock(test=Mock(name="test2", test_template=Mock()))],
+            [Mock(test=Mock(name="test3", test_template=Mock()))],
+            [
+                "SUCCESS_0=$(grep -q 'Avg bus bandwidth' stdout.txt && echo 1 || echo 0)",
+                "SUCCESS_1=$(grep -q 'Avg bus bandwidth' stdout.txt && echo 1 || echo 0)",
+                "PROLOGUE_SUCCESS=$( [ $SUCCESS_0 -eq 1 ] && [ $SUCCESS_1 -eq 1 ] && echo 1 || echo 0 )",
+                "if [ $PROLOGUE_SUCCESS -eq 1 ]; then",
+                "    srun",
+                "    epilogue",
+                "fi",
+            ],
+        ),
+    ],
+)
+def test_prologue_epilogue_combinations(
```

Comment:
This test and its parameters are very hard to read. Conditions could also be simplified as you are using Mocks here.

```python
"SUCCESS_0=cond1",
"SUCCESS_1=cond2",
"PROLOGUE_SUCCESS=$( [ $SUCCESS_0 -eq 1 ] && [ $SUCCESS_1 -eq 1 ] && echo 1 || echo 0 )",
"if [ $PROLOGUE_SUCCESS -eq 1 ]; then",
"    srun",
"    epilogue",
"fi",
```

And please restore acceptance testing, this covers only logic itself, but we do not want to loose in real coverage.

Reply:
> And please restore acceptance testing, this covers only logic itself, but we do not want to loose in real coverage.
I will do

> This test and its parameters are very hard to read. Conditions could also be simplified as you are using Mocks here.
I don’t think parameters are hard to read. Let’s discuss it in the meeting.